### PR TITLE
Create files for DogBreedList

### PR DESCRIPTION
**Pull request by qetiavanashvili**
https://github.com/ashalva/ios.memenator/issues/2

1. Create new files DogBreedsListView and DogBreedsListViewModel

2. Add new list item in main page which will open empty DogBreedsListView

### CODE REVIEW

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+struct DogBreedsListView: View {
+    var body: some View {
+        EmptyView()
+    }
+}
+
+struct DogBreedsListView_Previews: PreviewProvider {
+    static var previews: some View {
+        MainView(
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift]: `DogBreedsListView_Previews` should instantiate `DogBreedsListView` not MainView

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,9 @@
+//
+//  DogBreedsListViewModel.swift
+//  GeorgianMemenator
+//
+//  Created by Qeti Avanashvili on 08.01.23.
+//  Copyright © 2023 Shalva Avanashvili. All rights reserved.
+//
+
+import Foundation
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift:10]: Lets create the empty class for `DogBreedsListViewModel`, please check how `RedditViewModel` is created.

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import Foundation
+
+struct DogBreedsListView: View {
+    var body: some View {
+        EmptyView()
+    }
+}
+
+struct DogBreedsListView_Previews: PreviewProvider {
+    static var previews: some View {
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift:22]: Whenever you will create class `DogBreedsListViewModel`, we will need viewmodel property here.

`@ObservedObject var viewModel: DogBreedsListViewModel`

File: GeorgianMemenator/Scenes/MainView/MainViewModel.swift [hunk 4]
```diff
@@ -18,6 +18,7 @@ class MainViewModel: ObservableObject {
         MemeCategory(title: "Cat status codes", icon: Image("cat")),
         MemeCategory(title: "Girchi", icon: Image("cone")),
         MemeCategory(title: "References", icon: Image("hand")),
+        MemeCategory(title: "DogBreedsList", icon: Image("dog"))
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/MainView/MainViewModel.swift]: Please download the any dof file from here https://www.flaticon.com/search?word=dog

Download .svg file from there and add it to Assets and use that icon here.

**Comment by ashalva** [comment]
@qetiavanashvili you should put only yourself as assignees 👍

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift [hunk 5]
```diff
@@ -6,4 +6,21 @@
 //  Copyright © 2023 Shalva Avanashvili. All rights reserved.
 //
 
+import SwiftUI
 import Foundation
+
+class DogBreedsListViewModel: ObservableObject {
+    var body: some View {
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift]: Viewmodel does not have anything to do with views. So we need to delete the `body`. Lets leave this class empty for now.

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift [hunk 6]
```diff
@@ -6,4 +6,21 @@
...
+        EmptyView()
+        
+    }
+}
+
+struct DogBreedsListViewModel_Previews: PreviewProvider {
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift]: Same applies for this. Viewmodel does not have anything related to view :) Please delete entire `DogBreedsListViewModel_Previews`

File: GeorgianMemenator/Scenes/Reddit/RedditViewModel.swift [hunk 7]
```diff
@@ -9,6 +9,7 @@
 import Combine
 import Foundation
 
+
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/Reddit/RedditViewModel.swift]: Unneeded change, please delete that empty line

File: GeorgianMemenator/Assets.xcassets/Icons/dog.imageset/Contents.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "filename" : "dog.png",
```
Comment (ashalva) [inline GeorgianMemenator/Assets.xcassets/Icons/dog.imageset/Contents.json:4]: You have added `dog.png`, ideally we want to have `.svg` files. Let it be for now, but consider in the future. 

There is an option on flaticon to download .svg version of the icon.

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift (hunk 6 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift]: I have done...

File: GeorgianMemenator/Scenes/Reddit/RedditViewModel.swift (hunk 7 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Scenes/Reddit/RedditViewModel.swift]: I have done...

File: GeorgianMemenator/Assets.xcassets/Icons/dog.imageset/Contents.json (hunk 8 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Assets.xcassets/Icons/dog.imageset/Contents.json:4]: Screenshot 2023-01-10 at 22 25 52

There was just one option and what's why I downloaded dog.png file

Comment (ashalva) [reply GeorgianMemenator/Assets.xcassets/Icons/dog.imageset/Contents.json:4]: You should open its details, but its okay no worries

File: GeorgianMemenator/Scenes/Reddit/RedditViewModel.swift (hunk 7 above)
Comment (ashalva) [reply GeorgianMemenator/Scenes/Reddit/RedditViewModel.swift]: You deleted in `RedditView`, you should have deleted in `RedditViewModel`

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+//
+//  Created by Qeti Avanashvili on 08.01.23.
+//  Copyright © 2023 Shalva Avanashvili. All rights reserved.
+//
+
+import SwiftUI
+import Foundation
+
+class DogBreedsListViewModel: ObservableObject {
+
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift:13]: Please format this file. No new lines and unneeded spaces.

Comment (qetiavanashvili) [reply GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift:13]: I have corrected everything, hope there won't be any problem 🙏🙏

**Review by ashalva — APPROVED** [review]
Looking good ✅ Good to merge

**Comment by ashalva** [comment]
@qetiavanashvili since Giorgi has merged his pr first, you can not merge that easily now. Please, follow the steps:

1. `git checkout main`
2. `git pull`
3. `git checkout your-new-branch`
4. `git merge main`
5. `git checkout  --theirs GeorgianMemenator.xcodeproj/project.pbxproj`

After that you need to open Xcode. Your files will not be in XCode, so please right click on Scenes and add your folder.

image

After that you will need to fix the conflict in `MainView` and `MainViewModel`. Keep in mind that your changes and Giorgi's changes both should stay in those files.

After that make sure that the project builds and your and Giorgis changes are visible. Then push the changes like you have been doing.

**Pull request merged**